internal/stack/_static/docker-compose-stack.yml.tmpl

```
@@ -139,6 +139,9 @@ services:
       interval: 5s
     hostname: docker-fleet-agent
     env_file: "./elastic-agent.env"
+    {{- if eq (fact "agent_has_publish_ports") "true" }}
+    ports: {{ fact "agent_publish_ports" }}
+    {{- end }}
```

Nit. Maybe we can pass here the comma separated list of ports, without the brackets, and check here if the list is empty? This way we would not need the `has_publish_ports` fact.
```suggestion
    {{- if ne (fact "agent_publish_ports") "" }}
    ports: {{ fact "agent_publish_ports" }}
    {{- end }}
```